Декоративные элементы на главной странице

dobrjashka145 opened the pull request:
---
:mortar_board: [Постройте мелкие сетки на гридах для подходящих компонентов](https://up.htmlacademy.ru/htmlcss/33/user/106057/tasks/9)

---
:mortar_board: [Декоративные элементы на главной странице](https://up.htmlacademy.ru/htmlcss/33/user/106057/tasks/10)

dobrjashka145 committed:
Сверстана на гридах сетка для товаров в каталоге

dobrjashka145 committed:
Исправлен object-fit

dobrjashka145 committed:
Сделана верстка на флексах и гридах, стилизована часть декоративных элементов

andreysgra commented on index.html:
```diff
@@ -8,14 +8,14 @@
           <form class="search-form" action="https://echo.htmlacademy.ru/" method="post">
             <label class="visually-hidden" for="search-id">Поиск</label>
-            <input class="search-field" type="text" id="search-id" name="search-text" placeholder="Текст поиска">
+            <input class="search-field" type="text" id="search-id" name="search-text" placeholder="Поиск по сайту" disabled>
```
Хм-м, а почему `disabled`?

andreysgra commented on index.html:
```diff
@@ -89,7 +89,7 @@ <h2 class="visually-hidden">Слайдер товаров</h2>
               <h3 class="slider-title page-title">Порхает как бабочка, жалит как пчела!</h3>
               <p class="slider-desc">Этот обычный, на первый взгляд, квадрокоптер оснащен мощным лазером, замаскированным под стандартную камеру.</p>
-              <a class="slider-detals uppercase-spacing base-button" href="#">Подробнее</a>  
+              <button class="slider-detals uppercase-spacing base-button" href="#">Подробнее</button>  
```
Это ссылка, так как предполагается переход на другую страницу.
![image](https://user-images.githubusercontent.com/17790771/156170085-7c360577-0354-4dbf-8d1e-d9073b704036.png)
Первоначальный вариант разметки у вас был правильным.

dobrjashka145 replied:
Исправлено, как и аналогичные кнопки на карточках товара внутренней страницы

andreysgra commented on styles/styles.css:
```diff
@@ -69,13 +59,55 @@ body {
 }
 
+.uppercase-spacing {
+  font-family: Raleway, Arial,sans-serif;
```
Веб-безопасны шрифт (тот, что идёт вторым по списку) можно не указывать. Достаточно указания семейства шрифтов. Об этом есть упоминание в критерии [CSS-06](https://up.htmlacademy.ru/htmlcss/33/criteries#CSS-06)

Это не замечание, а просто ремарка.

andreysgra commented on styles/styles.css:
```diff
@@ -69,13 +59,55 @@ body {
+  outline: 2px solid #AF4FFF;  
+}
+
+.base-button:focus:hover {
+  background: #FFE17F;
+  outline: none;
+}
+
+.base-button:focus:active {
```
А зачем собственно `:focus:hover` и `:focus:active`? Если пользователю нужно взаимодействовать с элементом находящимся в фокусе, то нет необходимости убирать у него обводку, которая указывает, что данный элемент в фокусе.

dobrjashka145 replied:
Я не знала, как правильно отображать состояния с элементом в фокусе, ориентировалась на стайлгайд. Если не надо убирать обводку, это сильно упрощает жизнь. Исправляю

andreysgra commented on styles/styles.css:
```diff
@@ -110,13 +141,37 @@ body {
+  outline: 2px solid #AF4FFF;
+}
+
+.logo:focus:hover {
+  opacity: 0.6;
+  outline:none;
+}
+
+.logo:focus:active {
```
То же, что и для `base-button`.

dobrjashka145 replied:
Там, где остались focus:hover, focus:active, у меня по другому не получалось - для элемента в фокусе не отображалось то одно состояние, то другое.

andreysgra commented on styles/styles.css:
```diff
@@ -585,6 +704,49 @@ body {
+  box-shadow: 0 0 0 2px #ffffff, 0 0 0 4px #AF4FFF;
+}
+
+.mailing-button:focus:hover {
+  color: #FFFFFF;
+  background: #000000;
+  border: 3px solid #000000;
+  box-shadow: none;
+}
+
+.mailing-button:focus:active {
```
И здесь, как и для `base-button`.

dobrjashka145 committed:
стилизованы декоративные элементы на главной странице

Посылаю всем смертям , вражде и ненависти вопреки. Потому что основа и смысл жизни - любовь, творчество и созидание, что по сути - одно и то же. Об этом надо помнить сейчас, в это надо верить и уже сейчас думать, как все будет потом
